src/main/java/net/earthcomputer/clientcommands/command/AreaStatsCommand.java

```diff
@@ -0,0 +1,205 @@
+                chunks++;
+                final WorldChunk chunk = world.getChunk(minMaxMaxMax, minMaxChunk.getPos().z);
+                blocks = loop(16 * minMaxMaxMax, 16 * minMaxMaxMax + 15, (maxZ >> 4) * 16, maxZ, minY, maxY, blocks, blockPredicate, chunk, mutablePos);
+            }
+            for (int maxMinMaxMax = maxMinChunk.getPos().z + 1; maxMinMaxMax < maxMaxChunk.getPos().z; maxMinMaxMax++) {
+                chunks++;
+                final WorldChunk chunk = world.getChunk(maxMinChunk.getPos().x, maxMinMaxMax);
+                blocks = loop((maxX >> 4) * 16, maxX, 16 * maxMinMaxMax, 16 * maxMinMaxMax + 15, minY, maxY, blocks, blockPredicate, chunk, mutablePos);
+            }
+            for (int chunkX = minMinChunk.getPos().x + 1; chunkX < maxMinChunk.getPos().x; chunkX++) {
+                for (int chunkZ = minMinChunk.getPos().z + 1; chunkZ < minMaxChunk.getPos().z; chunkZ++) {
+                    chunks++;
+                    final WorldChunk chunk = world.getChunk(chunkX, chunkZ);
+                    blocks = loop(16 * chunkX, 16 * chunkX + 15, 16 * chunkZ, 16 * chunkZ + 15, minY, maxY, blocks, blockPredicate, chunk, mutablePos);
+                }
+            }
+        }
+
+        final long entities = StreamSupport.stream(world.getEntities().spliterator(), false)
+                .filter(entity ->
+                        entity.getX() >= minX && entity.getX() <= maxX &&
+                        entity.getZ() >= minZ && entity.getZ() <= maxZ &&
+                        entity.getY() >= minY && entity.getY() <= maxY)
+                .count();
+
+        Box box = new Box(minX, minY, minZ, maxX + 1, maxY + 1, maxZ + 1);
+        RenderQueue.addCuboid(RenderQueue.Layer.ON_TOP, box, box, 0xFFFF0000, 60 * 20);
+
+        long endTime = System.nanoTime();
+
+        sendFeedback("commands.careastats.output.1", chunks, endTime - startTime, (endTime - startTime) / 1000000);
+        sendFeedback("commands.careastats.output.2", blocks, (maxX - minX + 1) * (maxY - minY + 1) * (maxZ - minZ + 1));
+        sendFeedback("commands.careastats.output.3", entities);
+
+        return blocks;
+    }
+
+    private static int loop(int start1, int end1, int start2, int end2, int start3, int end3, int blocks, Predicate<Block> predicate, WorldChunk chunk, BlockPos.Mutable mutablePos) {
+        for (int x = start1; x <= end1; x++) {
+            for (int z = start2; z <= end2; z++) {
+                for (int y = start3; y <= end3; y++) {
+                    if (predicate.test(chunk.getBlockState(mutablePos.set(x, y, z)).getBlock())) {
+                        blocks++;
+                    }
+                }
+            }
+        }
+        return blocks;
+    }
+
+    private static Predicate<Block> or(List<Predicate<Block>> predicates) {
+        return predicates.stream().reduce(t -> false, Predicate::or);
+    }
```

Comment:
Guava has another useful method: `Predicates.or`

Reply:
I'm unsure how I'm supposed to use that 🤐.

Reply:
> Returns a predicate that evaluates to true if any one of its components evaluates to true. The components are evaluated in order, and evaluation will be "short-circuited" as soon as a true predicate is found. It defensively copies the array passed in, so future changes to it won't alter the behavior of this predicate. If components is empty, the returned predicate will always evaluate to false.

https://guava.dev/releases/19.0/api/docs/com/google/common/base/Predicates.html#or(com.google.common.base.Predicate...)

Reply:
Basically it does exactly what the function you've written does :P

Reply:
I don't see the point of that at all. Plus IntelliJ says:
`Guava's functional primitives can be replaced with Java API`.

Reply:
It means you don't need to write this function yourself. What does intellij suggest you replace guava's method with?

Reply:
Do you find this cleaner?
```java
final Predicate<Block> blockPredicate = Predicates.or(blockPredicates.toArray(new com.google.common.base.Predicate[0]));
```
Notice the fully qualified class.

Reply:
Resolving this as I couldn't find a way to use Guava's method.